osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -147,25 +160,31 @@ public override bool UpdateSubTree()
 
         private void updateClock()
         {
-            if (parentClock == null) return;
+            if (parentGameplayClock == null)
+            {
+                validState = false;
```

Comment:
This is going to infinitely loop in `UpdateSubTree()`:

```
parentGameplayClock = null; // No gameplay from BDL

isValid = true; // Beginning of loop
updateClock(); // isValid = false()
// base.UpdateSubTree() doesn't get called
// LoadComplete() doesn't get called ^
// loop exits
// repeat
```

Reply:
How about setting `parentGameplayClock = Clock` here instead of in `LoadComplete()`?